torchvision/transforms/autoaugment.py

```diff
@@ -108,25 +108,132 @@ def _get_transforms(  # type: ignore[return]
+    elif op_name == "Contrast":
+        img = F.adjust_contrast(img, 1.0 + magnitude)
+    elif op_name == "Sharpness":
+        img = F.adjust_sharpness(img, 1.0 + magnitude)
+    elif op_name == "Posterize":
+        img = F.posterize(img, int(magnitude))
+    elif op_name == "Solarize":
+        img = F.solarize(img, magnitude)
+    elif op_name == "AutoContrast":
+        img = F.autocontrast(img)
+    elif op_name == "Equalize":
+        img = F.equalize(img)
+    elif op_name == "Invert":
+        img = F.invert(img)
+    else:
+        raise ValueError("The provided operator {} is not recognized.".format(op_name))
+    return img
+
+
+class TrivialAugment(torch.nn.Module):
+    r"""Dataset-independent data-augmentation with TrivialAugment, as described in
+    `"TrivialAugment: Tuning-free Yet State-of-the-Art Data Augmentation" <https://arxiv.org/abs/2103.10158>`.
+        If the image is torch Tensor, it should be of type torch.uint8, and it is expected
+        to have [..., 1 or 3, H, W] shape, where ... means an arbitrary number of leading dimensions.
+        If img is PIL Image, it is expected to be in mode "L" or "RGB".
+
+        Args:
+            augmentation_space (str): A string defining which augmentation space to use.
+                The augmentation space can either set to be the one used for AutoAugment (`aa`)
+                or to the strongest augmentation space from the TrivialAugment paper (`ta_wide`).
+            num_magnitude_bins (int): The number of different magnitude values.
+            interpolation (InterpolationMode): Desired interpolation enum defined by
+                :class:`torchvision.transforms.InterpolationMode`. Default is ``InterpolationMode.NEAREST``.
+                If input is Tensor, only ``InterpolationMode.NEAREST``, ``InterpolationMode.BILINEAR`` are supported.
+            fill (sequence or number, optional): Pixel fill value for the area outside the transformed
+                image. If given a number, the value is used for all bands respectively.
+        """
+
+    def __init__(self, augmentation_space: str = 'ta_wide', num_magnitude_bins: int = 30,
+                 interpolation: InterpolationMode = InterpolationMode.NEAREST,
+                 fill: Optional[List[float]] = None):
+        super().__init__()
+        self.augmentation_space = augmentation_space
+        self.num_magnitude_bins = num_magnitude_bins
+        self.interpolation = interpolation
+        self.fill = fill
+
+    def forward(self, img: Tensor):
+        fill = self.fill
+        if isinstance(img, Tensor):
+            if isinstance(fill, (int, float)):
+                fill = [float(fill)] * F._get_image_num_channels(img)
+            elif fill is not None:
+                fill = [float(f) for f in fill]
+
+        op_meta = _get_magnitudes(self.augmentation_space, F._get_image_size(img), num_bins=self.num_magnitude_bins)
+        op_index = torch.randint(len(op_meta), (1,))
```

Comment:
There is a specific convention we use for transforms that requires adding all the random calls not directly in the `forward()` but instead on a separate static `get_params()` method. See AutoAugment's implementation as an [example](https://github.com/pytorch/vision/blob/master/torchvision/transforms/autoaugment.py#L166-L176).

Reply:
Oh, I see. This is of course easily fixable. :)